app/js/reducers/artifacts.js

```diff
@@ -23,8 +23,11 @@ const artifacts = (state = initialState, action) => {
         });
         return state.set(action.artifact.id, nextState);
     }
-    case 'DELETE_ARTIFACT':
-        return state.delete(action.id);
+    case 'DELETE_ARTIFACT': {
+        // TODO, make it actually do something
+        const newState = ImmutableMap(state);
+        return newState.delete(action.id).toJS();
```

Comment:
Yeah, so it technically is being used properly (which is why it passed testing before this hack), but it appears that using `combineReducers()` from redux breaks Immutable.js usability as it sends objects as straight JS to their individual reducers instead of keeping their Immutable data structure, like what would happen with using a single reducer.

We'll have to decide if we even want to continue using it, add yet another dependency to make it work naturally ([`redux-immutable`](https://github.com/gajus/redux-immutable)), or just use normal JS objects.